a-star_test.go

```diff
@@ -116,3 +119,240 @@ func ExampleShortestPathWithHeuristic() {
+
+		g.Connect(nodeKey, strconv.Itoa(right), rand.Intn(5))
+		g.Connect(nodeKey, strconv.Itoa(rightDown), rand.Intn(5))
+		g.Connect(nodeKey, strconv.Itoa(down), rand.Intn(5))
+		g.Connect(nodeKey, strconv.Itoa(downLeft), rand.Intn(5))
+	}
+
+	return g
+}
+
+func getRight(nodePos, columns int) int {
+	if nodePos% columns == 0 {
+		return nodePos - (columns - 1)
+	}
+	return nodePos + 1
+}
+
+func getLeft(nodePos, columns int) int {
+	if nodePos% columns == 1 {
+		return nodePos + (columns - 1)
+	}
+	return nodePos - 1
+}
+
+func getDown(nodePos, rows, columns int) int {
+	FirstNodeOfLastRow := rows*columns - columns + 1
+	if nodePos >= FirstNodeOfLastRow {
+		return nodePos - (rows*columns - columns)
+	}
+	return nodePos + columns
+}
+
+func BenchmarkShortestPathWithHeuristic_100X100GridOfNodes(b *testing.B) {
+	rows, columns := 100, 100
+	grid := createGrid(rows, columns)
+	randomNodes := createLongListOfRandomKeyNodes(rows * columns, 100000)
+
+	b.ResetTimer()
+
+	for i := 0; i < b.N; i++ {
+		_, ok := grid.ShortestPathWithHeuristic(randomNodes[i].fromNode, randomNodes[i].toNode, heuristicFor100X100Grid)
+		if !ok {
+			b.Fatal(`ok is false`, randomNodes[i])
+		}
+	}
+}
+
+type randFromTo struct {
+	fromNode, toNode string
+}
+
+func createLongListOfRandomKeyNodes(totalNumberOfNodes, resultLength int) (res []randFromTo) {
+	rand.Seed(time.Now().UnixNano())
+
+	for i:=0; i < resultLength; i++ {
+		randRes := randFromTo{"0", "0"}
+		for ; randRes.fromNode == "0" || randRes.toNode == "0" ; {
+			randRes = randFromTo{
+				fromNode: strconv.Itoa(rand.Intn(totalNumberOfNodes)),
+				toNode:   strconv.Itoa(rand.Intn(totalNumberOfNodes)),
+			}
+		}
+
+		res = append(res, randRes)
+	}
+
+	return res
+}
+
+func TestHeuristicCalculationFor100X100Grid(t *testing.T) {
+	expected := []struct {
+		startNode, endNode string
+		distance int
+	}{
+		{
+			startNode: "0",
+			endNode:   "1",
+			distance:  1,
+		},
+		{
+			startNode: "1",
+			endNode:   "3",
+			distance:  2,
+		},
+		{
+			startNode: "1",
+			endNode:   "101",
+			distance:  1,
+		},
+		{
+			startNode: "220",
+			endNode:   "450",
+			distance:  30,
+		},
+		{
+			startNode: "50",
+			endNode:   "850",
+			distance:  8,
+		},
+		{
+			startNode: "950",
+			endNode:   "102",
+			distance:  48,
+		},
+		{
+			startNode: "940",
+			endNode:   "7172",
+			distance:  38, // 32 to the right + 62 down (38 up)
+		},
+		{
+			startNode: "7172",
+			endNode:   "940",
+			distance:  38, // 32 to the left + 62 up (38 down)
+		},
+		{
+			startNode: "7178",
+			endNode:   "7920",
+			distance:  42, // 58 to the left >
+		},
+	}
+
+	for _, e := range expected {
+		res := heuristicFor100X100Grid(e.startNode, e.endNode)
+		if res != e.distance {
+			t.Fatalf("Distance between startNode %v and endNode %v should have been %v but was %v",
+				e.startNode, e.endNode, e.distance, res)
+		}
+	}
+}
+
+func heuristicFor100X100Grid(startKey, endKey string) int {
+	rows, cols := 100, 100
```

Comment:
I agree but at the current code state doing so will make the tests fail.

I can do if u want but it will need some time, how about doing it at the next pull request? ;)

Reply:
Why would that make the test fail? I don't understand. If it's okay, I'll do it and push to your branch?

Reply:
Currently the only function which test it is the TestHeuristicCalculationFor100X100Grid, u can switch to 100x100 but it will make the illusion that other calculations are valid, to make that valid it need more test functions.

Also I think that current calculations won't support non square grids, but I need to allocate a little time to make sure.

If u want to make changes then feel free, who am I to block u on making changes to ur own project? ;)

Reply:
Also, as said b4, I'll be happy to do it at some later time as well

Reply:
> but it will make the illusion that other calculations are valid

I don't think so. It will return a heuristic function that's valid for whatever grid sizes you pass into it. Of course, you have to use the grid size you test with when calling this function.

Reply:
Maybe it will but I can't allocate time for that now (changing the code & adding fitting tests), if u want feel free to go ahead and do the necessary changes or give me few days to do it, however u want.

Reply:
So what did we agree upon at the end?
u will do the update or me?

Reply:
Let's merge this for now, having any benchmark is better then having none at all. I might change some things, though, at a later time.